tools/ktx/metrics_utils.h

```diff
@@ -94,11 +94,22 @@ class MetricsCalculator {
         KTXTexture2 texture{static_cast<ktxTexture2*>(malloc(sizeof(ktxTexture2)))};
         ktxTexture2_constructCopy(texture, encodedTexture);
 
-        const auto tSwizzleInfo = determineTranscodeSwizzle(texture, report);
+		// Start with a default swizzle
+        TranscodeSwizzleInfo  tSwizzleInfo{};
+		tSwizzleInfo.defaultNumComponents = 4;
+		tSwizzleInfo.swizzle = "rgba";
 
         ktx_error_code_e ec = KTX_SUCCESS;
         // Decode the encoded texture to observe the compression losses
-        ec = ktxTexture2_TranscodeBasis(texture, KTX_TTF_RGBA32, 0);
+		if (isFormatAstc((VkFormat)texture->vkFormat))
+		{
+			ec = ktxTexture2_DecodeAstc(texture, VK_FORMAT_R8G8B8A8_UNORM);
```

Comment:
Please update the documentation for `--compare-ssim` and `--compare-psnr`. I failed to notice before that they  are currently only valid for BasisU compression. The documentation still says that.